engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -2246,9 +2246,6 @@ public Network createGuestNetwork(final long networkOfferingId, final String nam
                 if (_networkModel.areServicesSupportedByNetworkOffering(ntwkOff.getId(), Service.SourceNat)) {
                     throw new InvalidParameterValueException("Service SourceNat is not allowed in security group enabled zone");
                 }
-                if (!_networkModel.areServicesSupportedByNetworkOffering(ntwkOff.getId(), Service.SecurityGroup)) {
```

Comment:
Instead of removing the check, shouldn't we fix the `ntwkOff` object and related objects/attributes/configurations that are used to identify if the network supports security groups?

Reply:
I'm not sure I get what you are hinting at. I'm just removing the check that SG isn't mandatory.

Could you elaborate a bit more on what you mean?

Reply:
lgtm

Reply:
@wido it was my mistake. After checking this code with @GabrielBrascher, I now understood what you did.